pkg/minikube/node/start.go

```diff
@@ -334,35 +372,95 @@ func joinCluster(starter Starter, cpBs bootstrapper.Bootstrapper, bs bootstrappe
+		if starter.Node.OS != "windows" {
+			if err := bs.JoinCluster(*starter.Cfg, *starter.Node, joinCmd); err != nil {
+				// log the error message and retry
+				klog.Errorf("%s node failed to join cluster, will retry: %v", role, err)
+
+				// reset node to revert any changes made by previous kubeadm init/join
+				klog.Infof("resetting %s node %q before attempting to rejoin cluster...", role, starter.Node.Name)
+				if _, err := starter.Runner.RunCmd(exec.Command("/bin/bash", "-c", fmt.Sprintf("%s reset --force", bsutil.InvokeKubeadm(starter.Cfg.KubernetesConfig.KubernetesVersion)))); err != nil {
+					klog.Infof("kubeadm reset failed, continuing anyway: %v", err)
+				} else {
+					klog.Infof("successfully reset %s node %q", role, starter.Node.Name)
+				}
+
+				return err
+			}
+		} else {
+			driverIP, err := starter.Host.Driver.GetIP()
+			if err != nil {
+				klog.Errorf("Unable to get driver IP: %v", err)
 			}
+			klog.Infof("Driver IP: %s", driverIP)
+
+			// Call with a timeout of 30 seconds
+			timeout := 20 * time.Second
+
+			if commandResult, err := bs.JoinClusterWindows(starter.Host, *starter.Cfg, *starter.Node, joinCmd, timeout); err != nil {
+				klog.Infof("%s node failed to join cluster, will retry: %v", role, err)
+				klog.Infof("command result: %s", commandResult)
+
+				// sort out the certificates issues
+				if cmd, err := bs.SetupMinikubeCert(starter.Host); err != nil {
+					klog.Errorf("error setting minikube folder error script: %v", err)
+				} else {
+					klog.Infof("command result: %s", cmd)
+					// retry the join command
+					if commandResult, err := bs.JoinClusterWindows(starter.Host, *starter.Cfg, *starter.Node, joinCmd, 0); err != nil {
+						klog.Errorf("error retrying join command: %v, command result: %s", err, commandResult)
+						return err
+					}
 
-			return err
+					// set up flannel network issues
+					if err := prepareWindowsNodeFlannel(); err != nil {
+						klog.Errorf("error preparing windows node flannel: %v", err)
+					}
+
+					// set up kube-proxy issues
+					if err := prepareWindowsNodeKubeProxy(); err != nil {
+						klog.Errorf("error preparing windows node kube-proxy: %v", err)
+					}
+				}
+				// return err
+
+			}
 		}
 		return nil
 	}
 	if err := retry.Expo(join, 10*time.Second, 3*time.Minute); err != nil {
-		return fmt.Errorf("error joining %s node %q to cluster: %w", role, starter.Node.Name, err)
+		if starter.Node.OS != "windows" {
+			return fmt.Errorf("error joining %s node %q to cluster: %w", role, starter.Node.Name, err)
+		}
 	}
 
-	if err := cpBs.LabelAndUntaintNode(*starter.Cfg, *starter.Node); err != nil {
-		return fmt.Errorf("error applying %s node %q label: %w", role, starter.Node.Name, err)
+	if starter.Cfg.WindowsNodeVersion != "2022" {
+		if err := cpBs.LabelAndUntaintNode(*starter.Cfg, *starter.Node); err != nil {
+			return fmt.Errorf("error applying %s node %q label: %w", role, starter.Node.Name, err)
+		}
```

Comment:
why are we excluding server 2022 only from this block?

Reply:
Removing this bit of unrequired code.